### PR TITLE
Database ReLogic

### PR DESCRIPTION
**Pull request by WyrdWyn4**
### Tickets TT-71 & TT-81

# Changes
## 1. `Recursive Logic Functions`
 - Recursive Logic functions were used for many-to-many relationship table updates (If A updates, B must also update)
 - Fixed functionality _(Accidentally used conditional statement at the wrong place)_
## 2. `Auth Service`
 - Updated User struct
 - Renamed `user-repository.go` to `user-db.go` to ensure naming consistency throughout code.
## 3. `Main Service`
 - In addition to Recursive Logic Functions, added `UpdateName` function, from Profile Service
## 4. `Profile Service`
 - Updated Tables
## 5. `Search Service`
 - Updated Tables
 - Renamed `itinerary.go` to `itinterary-db.go` to ensure naming consistency throughout code.
 - Renamed `event.go` to `events-db.go` to ensure naming consistency throughout code.
## 6. `Column Changes`
 - Removed many-to-many relationship between `Itineraries` and `Events`
 - Updated `Itinerary`, `Post` and `Event` Columns
    - `Posts`:
          - Added Likes and Comments
          - Removed Title, Description, PostImages, Tags
    - `Itineraries`:
          - Added Title, Description, Price
          - Removed CreationDate and LastUpdate
    - `Events`:
          - Renamed Location to Address
          - Renamed StartDate to StartTime
          - Renamed EndDate to EndTime
          - Renamed Price to Cost, and made it a floating point value.
## 7. `Comments Table`
 - Added `Comments` table

# Notes
 - Ran all the services:
    - Main, Auth and Search services, work fine
    - No idea how to test Profile service _(I guess it is working, I created an account?)_
    - Please see below
 
Service Run Image

### CODE REVIEW

File: backend/services/itinerary-service/internal/db/itinerary-db.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+type Itinerary struct {
+	ItineraryId int      `json:"itineraryId"`
+	Name        string   `json:"name"`
+	City        string   `json:"city"`
+	Country     string   `json:"country"`
+	Title       string   `json:"title"`
+	Description string   `json:"description"`
+	Price       float64  `json:"price"`
+	Languages   []string `json:"languages"`
```
Comment (jordyob03) [inline backend/services/itinerary-service/internal/db/itinerary-db.go:19]: I don't know if we need to have languages in itineraries I feel like if there are any language barriers the user could just put it in the description.

File: backend/services/itinerary-service/internal/db/post-db.go [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+	"github.com/lib/pq"
+)
+
+type Post struct {
+	PostId       int       `json:"postId"`
+	ItineraryId  int       `json:"itineraryId"`
+	CreationDate time.Time `json:"dateOfCreation"`
+	Username     string    `json:"username"`
+	Tags         []string  `json:"tags"`
```
Comment (jordyob03) [inline backend/services/itinerary-service/internal/db/post-db.go:17]: If tags are stored in the itinerary do we need them here as well?

Comment (WyrdWyn4) [reply backend/services/itinerary-service/internal/db/post-db.go:17]: Good Point. We don't. I'll remove those in the post.

File: backend/services/itinerary-service/internal/db/itinerary-db.go (hunk 1 above)
Comment (WyrdWyn4) [reply backend/services/itinerary-service/internal/db/itinerary-db.go:19]: I don't have any stronger view on this tbh, and am good with any implementation. I'll ask the rest of the team.

**Review by jordyob03 — APPROVED** [review]
Looks good to me! Great job!

**Review by AbdulTur — APPROVED** [review]
LGTM

**Review by Leo-Gilbert — APPROVED** [review]
looks amazing!!!!!!

**Pull request merged**